lib/mapping/CCampaignHandler.cpp

```diff
@@ -151,6 +180,257 @@ std::string CCampaignHandler::readLocalizedString(CBinaryReader & reader, std::s
 	return VLC->generaltexth->translate(stringID.get());
 }
 
+CCampaignHeader CCampaignHandler::readHeaderFromJson(JsonNode & reader, std::string filename, std::string modName, std::string encoding )
+{
+	CCampaignHeader ret;
+
+	ret.version = reader["version"].Integer();
+	if(ret.version < CampaignVersion::VCMI_MIN || ret.version > CampaignVersion::VCMI_MAX)
+	{
+		logGlobal->info("VCMP Loading: Unsupported campaign %s version %d", filename, ret.version);
+		return ret;
+	}
+	
+	ret.version = CampaignVersion::VCMI;
+	ret.mapVersion = reader["campaignId"].Integer();
+	ret.name = reader["name"].String();
+	ret.description = reader["description"].String();
+	ret.difficultyChoosenByPlayer = reader["allowDifficultySelection"].Bool();
+	//skip ret.music because it's unused in vcmi
+	ret.filename = filename;
+	ret.modName = modName;
+	ret.encoding = encoding;

```

Comment:
OK to keep, but should be unused for vcmp - json's are always encoded as utf-8, no need to support/consider H3 language-specific encodings

Reply:
This field is unused for now and here just for consistency